spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -170,31 +182,86 @@ public interface OrderRepository extends CrudRepository<Order, Long> {
+
+
+*Scenario 1: Person Repository with findByAddressZipCode(ZipCode zipCode)*
+
+In this scenario, we have the following data models:
 
 [source,java]
 ----
-@Repository
-public interface OrderRepository extends CrudRepository<Order, Long> {
-
-  Stream<Order> findByAddressZipCode(ZipCode zipCode);
+class Person {
+  private Long id;
+  private Address address;
+}
 
+class Address {
+  private Zipcode zipcode;
 }
 ----
 
+- The query method `findByAddressZipCode` takes a `ZipCode` object as a parameter.
+- The Property Resolution Algorithm will parse the method name and extract `AddressZipCode`.
+- It will then attempt to resolve the property named `addressZipCode` in the `Person` class, following automatic class splitting by camel case.
+- Since the `Person` class has an `Address` property, it will recursively follow the path to the `Address` class.
+- In the `Address` class, it will identify the `Zipcode` property and filter `Person` records based on the provided `ZipCode` object within the `Address` object.
+
+*Scenario 2: People Repository with findByAddressZipCode(String addressZipCode)*
+
 
-Although this should work for most cases, to resolve this ambiguity, you can use `_` inside your method name to manually define traversal points.
+In this scenario, we have the following data model:
 
 [source,java]
 ----
-@Repository
-public interface OrderRepository extends CrudRepository<Order, Long> {
+class Person {
+  private Long id;
+  private String addressZipCode;
+}
+----
 
-  Stream<Order> findByAddress_ZipCode(ZipCode zipCode);
+- The query method `findByAddressZipCode` takes a `String` parameter named `addressZipCode`.
+- The Property Resolution Algorithm will parse the method name and extract `AddressZipCode`.
+- It will then attempt to resolve the property named `addressZipCode` in the `Person` class, following automatic class splitting by camel case.
+- If a property named `addressZipCode` of type `String` exists in the `Person` class or its nested objects, the query will filter `Person` records based on the provided `addressZipCode` string.
 
+*Scenario 3: OrderRepository` Repository with `findByAddress_ZipCode(ZipCode zipCode)*
+
+In this scenario, we have the following data models:
+
+[source,java]
+----
+class Order {
+  private Long id;
+  private String addressZipCode;
+  private Address address;
+}
+
+class Address {
+  private Zipcode zipcode;
 }
 ----
 
+- The query method `findByAddress_ZipCode` takes a `ZipCode` object as a parameter.
+- The method name includes an underscore (`_`) indicating manual traversal points.
+- The Property Resolution Algorithm will parse the method name and extract `Address_ZipCode`, recognizing the underscore as a traversal point.
+- It will then attempt to resolve the property named `Address` within the `Order` class, followed by the `ZipCode` property within the `Address` class, following manual traversal points.
```

Comment:
```suggestion
- It will then attempt to resolve the property named `Address` within the `Order` class, followed by the `zipcode` property within the `Address` class, following manual traversal points.
```